### ISSUE TITLE
Add support for cookie retrieval

### ISSUE THREAD

**Issue by johnnycardy** [windows-toolkit/Microsoft.Toolkit.Win32#120]
I'm submitting a feature request (UserVoice request: https://wpdev.uservoice.com/forums/110705-universal-windows-platform/suggestions/37626457-webview-allow-retrieval-of-cookies).

The requirement is to access the browser cookies. In my specific scenario, I have a webview to display my online web app UI. Then, separately in C#, I want to make authenticated HTTP requests.

I can use this code to request cookies:
`webView.InvokeScriptAsync("eval", "document.cookie");` 
However, this doesn't work for secure/HTTP-only cookies.

**Comment by verelpode** [comment]
I agree that it would be useful if `WebView` (both `Microsoft.Toolkit.Wpf.UI.Controls.WebView` and `Windows.UI.Xaml.Controls.WebView`) supported an API with these capabilities:
* Get a listing of all cookies.
* Retrieve/read a specified cookie.
* Modify a preexisting cookie.
* Create a new cookie.
* Delete a specified cookie.
* Delete all cookies.

In our enterprise situation, we could make our systems run in a much more user-friendly manner if the software had full access to read/modify/delete cookies.   Although some old cookie API's already exist, I believe these don't work inside the UWP sandbox AFAIK.  It would be great if `WebView` had a clean and straightforward API for this purpose; an API that is guaranteed to be compatible with `WebView`, as opposed to old stuff that might only work with Internet Explorer and might break in a future version of Windows.

**Comment by verelpode** [comment]
Here is a concrete example of why we need cookie retrieval:  We are required to download + display PDF files from websites that only authorize the PDF downloads if the correct temporary/session cookie is available, but WebView runs in a separate process, thus when our app does a HTTP download, it doesn't see the temporary cookies in the WebView process, thus the website refuses to provide the PDF file.

Our app subscribes to `WebView.UnviewableContentIdentified` and then uses `Windows.Web.Http.HttpClient` (or `System.Net.Http.HttpClient`) to download the PDF file, but the website denies it because of the missing cookie.  We need a way of giving `HttpClient` access to the same cookies as `WebView` despite the fact they're running in different processes.  Thus if WebView allowed us to retrieve the cookies, then we could copy them for use with `HttpClient`.

See also `WebViewControlWebResourceRequestedEventArgs.Request.Headers.Cookie` but unfortunately it is not yet supported in WPF WebView, but it looks trivially easy to support it if `WebViewControlWebResourceRequestedEventArgs.Request` is simply changed to type `object`.
